Zend/zend_execute.c

```diff
@@ -4354,6 +4355,21 @@ static zend_always_inline uint32_t zend_get_arg_offset_by_name(
 				*(uintptr_t *)(cache_slot + 1) = i;
 				return i;
 			}
+
+			if (fbc->common.attributes != NULL) {
+				zend_attribute *attribute = zend_get_parameter_attribute_str(
```

Comment:
It looks like it'd be a bit more efficient to do this outside of this loop and find the parameters where `attr->offset > 0` - e.g. if there was a function with 20 parameters with attributes, it'd loop over all 20 of them just to find the ones at position `i`

The index to use would be `attr->offset - 1`

```c
static zend_attribute *get_attribute_str(HashTable *attributes, const char *str, size_t len, uint32_t offset)
{
	if (attributes) {
		zend_attribute *attr;

		ZEND_HASH_FOREACH_PTR(attributes, attr) {
			if (attr->offset == offset && ZSTR_LEN(attr->lcname) == len) {
				if (0 == memcmp(ZSTR_VAL(attr->lcname), str, len)) {
					return attr;
				}
			}
		} ZEND_HASH_FOREACH_END();
	}

	return NULL;
}

ZEND_API zend_attribute *zend_get_parameter_attribute(HashTable *attributes, zend_string *lcname, uint32_t offset)
{
	return get_attribute(attributes, lcname, offset + 1);
}
```